woodwork/column_accessor.py

```diff
@@ -165,15 +164,20 @@ def semantic_tags(self):
     def __eq__(self, other):
         if self._schema != other._schema:
             return False
+        if self._series.name != other._series.name:
+            return False
```

Comment:
Our test for this was comparing a datetime series to a categorical series, and wasn't actually checking whether the names being the only difference would cause two column's to not be equal, and it turns out it wasn't (I guess `series.equals(other_series)` doesn't catch different names?)